software/contrib/knob_playground.py

```diff
@@ -0,0 +1,70 @@
+from machine import ADC
+from time import sleep
+
+from europi import k1, k2, b1, b2, oled
+from europi_script import EuroPiScript
+from experimental.knobs import KnobBank
+
+"""
+An example program showing the use of a KnobBank or LockableKnobs. This script is not meant to be merged into main, 
+it only exists so that PR reviewers can try out a LockableKnob in physical hardware easily.
+"""
+
+
+class KnobPlayground(EuroPiScript):
+    def __init__(self):
+        super().__init__()
+        self.next_k1 = False
+        self.next_k2 = False
+
+        self.kb1 = (
+            KnobBank.builder(k1)
+            .with_locked_knob("p1", initial_value=1, threshold=0.02)
+            .with_locked_knob("p2", initial_value=1)
+            .with_locked_knob("p3", initial_value=1)
+            .build()
+        )
+        self.kb2 = (
+            KnobBank.builder(k2)
+            .with_disabled_knob()
+            .with_locked_knob("p4", initial_value=1, threshold=1 / 7)
```

Comment:
[discussion] It looks like `threshold` is an optional ratio relative to the max number of choices? What if instead we had a parameter like `max_choices` that handled the mathy stuff internally?

```
.with_locked_knob("p4", initial_value=1, max_choices=7)

...
# LockableKnob.__init__(...)
self.threshold = int((1/max_choices) * MAX_UINT16)
```

Reply:
Threshold is just a percentage. (maybe the parameter name should be `threshold_percentage`?) Here I'm using the number of choices to calculate a percentage that 'feels right'. The `LockedKnob` just like the regular `Knob` doesn't care about what method is used to read the position (choice(), percent(), etc) so I'd like to avoid going down a setup road that makes the user think that the knob can only be read in one way.

That said, this does sound really convenient.